Make sure we can autoreconfigure popt

This commit makes a few changes in the popt build system to allow it
to be autoreconfigured with modern autotools:

 - the TESTS variable shouldn't be using $(top_srcdir), and since we
   don't care about tests in Buildroot, we just disable it.

 - the AM_C_PROTOTYPES macro is deprecated and should no longer be
   used, causes an error with autoconf/automake.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: b/Makefile.am
===================================================================
--- a/Makefile.am
+++ b/Makefile.am
@@ -37,8 +37,6 @@
 TESTS_ENVIRONMENT = \
 test1="$(top_builddir)/test1"
 
-TESTS = $(top_srcdir)/testit.sh
-
 include_HEADERS = popt.h
 
 usrlibdir = $(libdir)
Index: b/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -46,7 +46,6 @@
 AC_SYS_LARGEFILE
 
 AC_ISC_POSIX
-AM_C_PROTOTYPES
 
 AC_CHECK_HEADERS(float.h fnmatch.h glob.h langinfo.h libintl.h mcheck.h unistd.h)
 
